Sources/ArgumentParser/Parsable Properties/Flag.swift

```diff
@@ -240,15 +242,118 @@ extension Flag where Value: CaseIterable, Value: RawRepresentable, Value.RawValu
+    self.init(_parsedValue: .init { key in
+      // This gets flipped to `true` the first time one of these flags is
+      // encountered.
+      var hasUpdated = false
+      
+      let args = Element.allCases.map { value -> ArgumentDefinition in
+        let caseKey = InputKey(rawValue: String(describing: value))
+        let name = Element.name(for: value)
+        let help = ArgumentDefinition.Help(options: .isOptional, help: help, key: key)
+        return ArgumentDefinition.flag(name: name, key: key, caseKey: caseKey, help: help, parsingStrategy: .nextAsValue, initialValue: nil as Element?, update: .nullary({ (origin, name, values) in
+          if hasUpdated && exclusivity == .exclusive {
+            throw ParserError.unexpectedExtraValues([(origin, String(describing: value))])
+          }
+          if !hasUpdated || exclusivity == .chooseLast {
+            values.set(value, forKey: key, inputOrigin: origin)
+          }
+          hasUpdated = true
+        }))
+      }
+      return exclusivity == .exclusive
+        ? ArgumentSet(exclusive: args)
+        : ArgumentSet(additive: args)
+      })
+  }
+  
+  /// Creates an array property that gets its values from the presence of
+  /// zero or more flags, where the allowed flags are defined by a
+  /// case-iterable type.
+  ///
+  /// This property has an empty array as its default value.
   ///
-  /// This property has a default value of `nil`; specifying the flag in the
-  /// command-line arguments is not required.
+  /// - Parameters:
+  ///   - name: A specification for what names are allowed for this flag.
+  ///   - help: Information about how to use this flag.
+  public init<Element>(
+    help: ArgumentHelp? = nil
+  ) where Value == Array<Element>, Element: EnumerableFlag {
+    self.init(_parsedValue: .init { key in
+      let args = Element.allCases.map { value -> ArgumentDefinition in
+        let caseKey = InputKey(rawValue: String(describing: value))
+        let name = Element.name(for: value)
+        let help = ArgumentDefinition.Help(options: .isOptional, help: help, key: key)
+        return ArgumentDefinition.flag(name: name, key: key, caseKey: caseKey, help: help, parsingStrategy: .nextAsValue, initialValue: [Element](), update: .nullary({ (origin, name, values) in
+          values.update(forKey: key, inputOrigin: origin, initial: [Element](), closure: {
+            $0.append(value)
+          })
+        }))
+      }
+      return ArgumentSet(additive: args)
+    })
+  }
+}
+
+// - MARK: Deprecated CaseIterable/RawValue == String
+
+extension Flag where Value: CaseIterable, Value: RawRepresentable, Value.RawValue == String {
+  /// Creates a property that gets its value from the presence of a flag,
+  /// where the allowed flags are defined by a case-iterable type.
   ///
   /// - Parameters:
   ///   - name: A specification for what names are allowed for this flag.
+  ///   - initial: A default value to use for this property. If `initial` is
```

Comment:
Drive by - "default", not "initial", right?  The second sentence also doesn't seem right.

Reply:
Parameter names are used for documentation, not the argument labels, so `initial` is correct here. How would you word the second sentence? Would something like "If `initial` is `nil`, the flag declared with this `@Flag` attribute is not required by the user." make more sense?

Reply:
Re param names - odd, for doc generation I thought it was the public name that was always listed.  https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/SingleLineComment.html seems to imply that since it shows the name listed in the generated docs.

Second sentence - yea, the fact that it just said _flag_ was what threw me off, I was thinking it was referring to another parameter or something.

Reply:
Right, for that declaration the parameter name and argument label are the same. We always use the parameter name since some parameters don’t have argument labels (e.g. `func foo(_ v: Int)`).

I’ll revise that second sentence throughout, I think we’re using it in a pretty standard way. Thanks for the feedback!